pkg/auth/auth.go

```diff
@@ -219,16 +284,19 @@ func Logout(systemContext *types.SystemContext, opts *LogoutOptions, args []stri
 		if !opts.AcceptUnspecifiedRegistry {
 			return errors.New("please provide a registry to logout from")
 		}
-		if server, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
+		if key, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
 			return err
 		}
-		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
+		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", key)
 	}
 	if len(args) != 0 {
 		if opts.All {
 			return errors.New("--all takes no arguments")
 		}
-		server = getRegistryName(args[0])
+		key, registry, ref, err = parseRegistryArgument(args[0], opts.AcceptRepositories)
+		if err != nil {
+			return errors.Wrap(err, "parse registry argument on logout")
```

Comment:
Non-blocking: “parse”, “on logout” is not all that helpful in this context. Decide whether it’s the responsibility of `parseRegistryArgument` or the caller to include the argument in the message (currently it is inconsistent), and then have the two cooperate to produce a good error message.

We can polish error messages any other time, ad infinitum, though…

Reply:
Just returned the `err` here for now.